File: trunk/uealib/src/taboosearch/Context.java
package taboosearch;

public class Context extends core.Context {
		
	private static Context instance;

	/**
	 * Возвращает экземпляр контекста.
	 */
	public static Context getInstance() {
		if (instance == null) {
			instance = new Context();
		}
		return instance;
	}
	
	Evaluator e;
	
}

File: taboosearch/src/taboosearch/Context.scala
package taboosearch
import core._
import scala.annotation.unchecked.uncheckedVariance

class Context[+S <: Solution](evaluator : Evaluator[S],
							  val taboolator : Taboolator[S],
							  bestSolutionEver : S @uncheckedVariance,
							  bestSolutionEverFitness : Double)
							  extends core.Context(evaluator, bestSolutionEver, bestSolutionEverFitness) {

	override def tick() = {
		super.tick();
		taboolator.tick();
	}
	
}